### ISSUE TITLE
FR: Firebase Phone Auth for App Clips

### ISSUE THREAD

**Issue by nidegen** [firebase/firebase-ios-sdk#8281]
### Feature proposal

* Firebase Component: Auth

As it seems, App Clips does not support opening URL's or registering for push notifications. This prevents the use of Phone Auth if I understand correctly. Is there any way to solve this easily?

Alternatively, it would be nice to have a way to transfer an anonymous auth to the main app once downloaded. I think shared keychain is unavailable for app clips, which makes this also very difficult.

**Comment by nidegen** [comment]
@ryanwilson should I merge this with #8221 ? Or is there any value at all in keeping this open if there is no activity?

**Comment by ryanwilson** [comment]
Hey @nidegen, apologies for the delay in response. I think it's worth keeping them separate to be able to track interest across both platforms, and they will require different solutions (i.e. if we make App Clips works that won't solve the macOS/tvOS availability and vise versa).

Thanks!

**Comment by asloup** [comment]
Yeah, our team would also love to see this feature.  We're going to have to re-architect things now, because we're implementing an App Clip and can't use the same flow (which includes Firebase Auth phone auth).

**Comment by danpecher** [comment]
Are there any updates with this issue? 🙏

**Comment by marchy** [comment]
As a heads up, the barebones Firebase SDK is giant at over 12MB+ uncompressed.

Given the App Clip total size limit of 15MB this is a major deal-breaker – ie: the library would have to come in at < 1MB to be feasible.

**Comment by paulb777** [comment]
The Firebase SDK is a set of several of SDKs, most of which are much smaller.
